src/runtime/app.tsx

```diff
@@ -40,7 +41,11 @@ export function App({ helmetContext }: { helmetContext?: object }) {
     async function refetchData() {
```

Comment:
`refetchData` will only be executed after first render, so in first render process will still need to set the theme.I recommend to move these logic to Appearance component.